tests_emacs.clj

```diff
@@ -0,0 +1,18 @@
+(ns tests-emacs
+  (:require [babashka.tasks :as tasks]
+            [clojure.string :as str]))
+
+(def install-buttercup-routine
+  '(progn
+    (require (quote package))
+    (add-to-list (quote package-archives)
+                 (quote ("melpa-stable" . "http://stable.melpa.org/packages")) t)
+    (package-install (quote buttercup))))
+
+(defn install-buttercup []
+  (let [cmd (-> install-buttercup-routine str (str/escape {\" "\\\""}))]
+    (tasks/shell (format "emacs -Q --batch --eval \"%s\"" cmd))))
```

Comment:
I think using single quotes here works too, without the escaping:

```
(shell (format "emacs -Q --batch --eval '%s'" cmd))
```

Have you tried that?

Reply:
Oh, so the reason it turned out to be like that, because initially the routine was all strings, and I wanted to use double quotes for the shell command, and keep single quotes for quoting sexps. But then I realized, I'm passing strings between two Lisp dialects, and they don't have to start as strings.

I think the routine list can be prefixed with a back-quote (instead of a single), and then every `(quote` replaced with single quotes. In that case we'd have to do the escaping, for sure. But maybe using back-quote outside of a macro would be confusing, I dunno.

Reply:
Oh right.